relay-event-normalization/src/normalize/mod.rs

```diff
@@ -918,6 +941,48 @@ impl Default for LightNormalizationConfig<'_> {
             span_description_rules: Default::default(),
             geoip_lookup: Default::default(),
             enable_trimming: false,
+            dynamic_measurements_config: None,
+        }
+    }
+}
+
+/// Container for global and project level [`MeasurementsConfig`]. The purpose is to handle
+/// the merging logic.
+#[derive(Clone, Debug)]
+pub struct DynamicMeasurementsConfig<'a> {
+    project: Option<&'a MeasurementsConfig>,
+    global: Option<&'a MeasurementsConfig>,
+}
+
+impl<'a> DynamicMeasurementsConfig<'a> {
+    /// Constructor for [`DynamicMeasurementsConfig`].
+    pub fn new(
+        project: Option<&'a MeasurementsConfig>,
+        global: Option<&'a MeasurementsConfig>,
+    ) -> Self {
+        DynamicMeasurementsConfig { project, global }
+    }
+
+    /// Returns an iterator over the union of [`BuiltinMeasurementKey`]s from both the global and
+    /// project-level [`MeasurementsConfig`]s, with the elements from project config coming first.
+    pub fn builtin_measurement_keys(&'a self) -> impl Iterator<Item = &'a BuiltinMeasurementKey> {
+        let from_project = self.project.iter().flat_map(|c| &c.builtin_measurements);
+        let from_global = self.global.iter().flat_map(|c| &c.builtin_measurements);
+
+        from_project.chain(from_global).unique()
+    }
+
+    /// Gets the max custom measurements value from the [`MeasurementsConfig`] from project level or
+    /// global level. If both of them are available, it will choose the most restrictive.
+    pub fn max_custom_measurements(&'a self) -> Option<&'a usize> {
+        match (&self.project, &self.global) {
+            (None, None) => None,
+            (None, Some(global)) => Some(&global.max_custom_measurements),
+            (Some(project), None) => Some(&project.max_custom_measurements),
+            (Some(project), Some(global)) => Some(min(
```

Comment:
Ideally, remove the import and qualify `std::cmp::min` directly or use the `min` method: `Some(a.min(b))`